meson.build

```diff
@@ -1,86 +1,177 @@
-project('Little-CMS', 'c',
+project(
+  'Little-CMS',
+  'c',
   version: '2.14',
-  meson_version: '>=0.48.0',
+  meson_version: '>=0.60.0',
```

Comment:
I need 0.60 because of the TIFF and JPEG lookup, as neither have a sane name with which to look them up.

For instance, libtiff is `libtiff-4` if asked through pkg-config, `libtiff` if it's a Meson wrap, or `TIFF` if it's CMake.

And libjpeg is the same, `jpeg` if it's the original, `libjpeg` if it's libjpeg-turbo and asked through its wrap or pkg-config. (This one has the added complexity of verifying the API version.)

Reply:
Sounds like a bug in the wrap. It should match the pkgconfig name.

Reply:
ping @eli-schwartz

Reply:
> Sounds like a bug in the wrap. It should match the pkgconfig name.

If built through CMake (the official way) it generates TiffTargets.cmake and libtiff-4.pc.

Reply:
> For instance, libtiff is `libtiff-4` if asked through pkg-config, `libtiff` if it's a Meson wrap, or `TIFF` if it's CMake.

@amyspark for the record, this is a serious WrapDB policy violation if true -- it's flat-out forbidden.

The WrapDB's entire *reason* for existing, is to provide lookup-name parity with the official upstream project.

> Sounds like a bug in the wrap. It should match the pkgconfig name.

@neheb fortunately, it's not a WrapDB bug. The WrapDB specifies a `[provide]` section already that defines `libtiff-4 = libtiff4_dep`, and it is this that Meson uses to resolve the wrap.

`dependency('libtiff')` will fail, because it will not find any subproject providing such a dependency. As a last-ditch heuristic, it will attempt to configure `subproject('libtiff')` and check if it overrides `libtiff`, fail (because the libtiff wrap doesn't override any such thing) and, as a result of that failure, discard the libtiff subproject as unusable.

@amyspark at a minimum, you MUST remove `libtiff` from the lookup of tiff_dep, it will not do anything. The `tiff` one won't either, I'm pretty sure, because it needs to be uppercased to match CMake. In fact, libtiff itself doesn't install a cmake find_package configuration at all. (It does provide package exports, but not find_package anything. Darnedest thing I ever did see.) There's a totally unofficial FindTIFF.cmake shipped as part of the cmake sources.

I am curious if you actually tested this on a system without pkg-config but with cmake? Which makes me wonder why this project needs to support some totally unofficial ad-hoc cmake.org thing when you were happy with the version that didn't work. Maybe that means pkg-config is good enough?

I reiterate: upstream libtiff doesn't provide cmake find_package support anyway. All they provide is:
- pkg-config: `dependency('libtiff-4', method: 'pkg-config')`
- something which claims to be useful inside cmake's FetchContent exclusively

If you want libtiff upstream to provide cmake find_package support, you'll need to open a gitlab MR asking for it -- if you do so, please implement it via the name `find_package(libtiff-4)` to match the pkg-config file. :D

Reply:
> And libjpeg is the same, jpeg if it's the original, libjpeg if it's libjpeg-turbo and asked through its wrap or pkg-config. (This one has the added complexity of verifying the API version.)

I'm interested in this claim too -- I just downloaded the IJG edition of libjpeg, and checked -- they provide the pkg-config name `libjpeg.pc`, just like libjpeg-turbo does.

Funny story: IJG only started providing a pkg-config file since 2018, two years after libjpeg-turbo started doing so.

Furthermore, libjpeg-turbo provides a cmake configuration only for `dependency('libjpeg-turbo', method: 'cmake')`, and IJG provides nothing at all.

Totally unofficial cmake.org stuff provides `FindJPEG.cmake`, which again needs capitalization that you did NOT add and perhaps did not test? and isn't the official way to look it up anyway.

I don't think you *can* look for libjpeg-turbo at all though, since that exposes a different ABI -- the `tj*` symbols instead of the `jpeg_*` ones. You don't want to find that at all via pkg-config, and you want to specify `modules: ` to cmake to get the correct library, so you'd have to first check for the generic pkg-config name, and then afterward check for -turbo via cmake only.

This could be handled, more or less, by adding a libjpeg dependency class to Meson and having it handle getting the cmake name correct. Projects like lcms would benefit automatically simply by optionally choosing to build them with a newer version of Meson.

Reply:
> I don't think you can look for libjpeg-turbo at all though, since that exposes a different ABI

I'm *not* looking for libjpeg-turbo, just for the libjpeg library it provides. However, said pkg-config file does *not* provide the correct (API) version in its file, hence the additional checks needed, which I ported from the Autotools script:

```
❯ cat /usr/local/lib/pkgconfig/libjpeg.pc
prefix=/usr/local/Cellar/jpeg-turbo/2.1.4
exec_prefix=/usr/local/Cellar/jpeg-turbo/2.1.4
libdir=/usr/local/Cellar/jpeg-turbo/2.1.4/lib
includedir=/usr/local/Cellar/jpeg-turbo/2.1.4/include

Name: libjpeg
Description: A SIMD-accelerated JPEG codec that provides the libjpeg API
Version: 2.1.4
Libs: -L${libdir} -ljpeg
Cflags: -I${includedir}
```

Reply:
> However, said pkg-config file does not provide the correct (API) version in its file, hence the additional checks needed, which I ported from the Autotools script:

libjpeg-turbo was created by forking libjpeg 6b, so all versions of libjpeg-turbo are good enough. I suspect the checks ported from the autotools script are intended to detect IJG libjpeg 5 -- because 6b was released the same year LittleCMS was created.

Since that was back in 1998 I'm not sure how important it is to detect and properly error out in such a situation. :)